app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -33,27 +39,59 @@ class WireActivityViewModel @Inject constructor(
 
     private val isUserLoggedIn = currentSession != null
     var serverConfig: ServerConfig = ServerConfig.DEFAULT
+    var ssoDeepLinkResult: DeepLinkResult.SSOLogin? = null
```

Comment:
Hmm, so when it's updated then in `WireActivity` `recreate()` is called which effectively closes and creates this activity from scratch, do I understand this correctly?
Couldn't we define it as a State of Flow to just recompose when the value changes instead of recreating the whole Activity or such thing isn't possible?

Reply:
Yes, it get updated when the intent [onNewIntent] gets called.

Reply:
But wouldn't using the recompose mechanism be more efficient than recreating the whole activity?

Reply:
i'm not sure that I get your point, let's take it online.